lib/connectors/mongodb/connector.rb

```diff
@@ -73,23 +73,21 @@ def do_health_check
       end
 
       def with_client
-        client = Mongo::Client.new(
-          @host,
-          database: @database,
-          app_name: 'Elastic Enterprise Search',
-          max_pool_size: 8,
-          socket_timeout: 60,
-          heartbeat_frequency: 120,
-          connect_timeout: 30,
-          direct_connection: @direct_connection
-        )
-
-        if @user.present? && @password.present?
-          client = client.with(
-            user: @user,
-            password: @password
-          )
-        end
+        client = if @user.present? || @password.present?
```

Comment:
```suggestion
        client = if @user.present? && @password.present?
```

Reply:
Unsure if this was intentional, but wanted to flag it in case

Reply:
Indeed, it was intentional though slightly unrelated to memory leak - we can still attempt to create a client for mongo when user provided only username or only password, and mongodb will raise an error itself, which we'll show to the user - a bit of a lazy validation of username/password on my side, can improve on that for sure!